source/reference/ejson/stringify.txt

```diff
@@ -0,0 +1,506 @@
+   {"_id":{"$oid":"64da90c1175f5091debcab28"},"custId":486,"purchaseDate":{"$date":"2023-08-01T00:00:00Z"},"quantity":9,"cost":{"$numberDecimal":"200.53"}}
+
+The single-line output format is convenient for scripting. 
+``EJSON.stringify()`` can also produce human-readable formatting:
+
+.. code-block:: shell
+
+   # Note: This example is formatted to fit on the page.
+
+   mongosh --quiet \
+           --eval "db.sales.find().forEach( \
+                      o => print( EJSON.stringify(o, null, 3 ) ) )"
+
+The output is: 
+
+.. code-block:: json
+   :copyable: false 
+
+   # Note: This is only the first document.
+
+   {
+      "_id": {
+         "$oid": "64da90c1175f5091debcab26"
+      },
+      "custId": 345,
+      "purchaseDate": {
+         "$date": "2023-07-04T00:00:00Z"
+      },
+      "quantity": 4,
+      "cost": {
+         "$numberDecimal": "100.60"
+      }
+   }
+
+- ``o`` is the BSON value that ``EJSON.stringify()`` converts on each
+  iteration of ``.forEach()``.
+- ``null`` is a place holder for an optional ``replacer``.  When the
+  ``replacer`` is absent, ``EJSON.stringify()`` returns all fields that
+  have a defined value.
+- ``3`` is the ``spacer`` value. It tells ``EJSON.stringify()`` to
+  indent each new level by 3 spaces. 
+
+If the you want the output string to have additional type information,
+add the ``{ relaxed: false }`` option:
+
+.. code-block:: shell
+
+   # Note: This example is formatted to fit on the page.
+
+   mongosh --quiet \
+           --eval "db.sales.find().forEach( \
+                 o => print( \
+                     EJSON.stringify( o, null, 3, { relaxed: false } ) \
+              ) )"
+
+The output is: 
+
+.. code-block:: json
+   :copyable: false 
+
+   # Note: This is only the first document.
+
+   {
+      "_id": {
+         "$oid": "64da90c1175f5091debcab26"
+      },
+      "custId": {
+         "$numberInt": "345"
+      },
+      "purchaseDate": {
+         "$date": {
+            "$numberLong": "1688428800000"
+         }
+      },
+      "quantity": {
+         "$numberInt": "4"
+      },
+      "cost": {
+         "$numberDecimal": "100.60"
+      }
+   }
+
+Filter Output Fields
+~~~~~~~~~~~~~~~~~~~~
+
+``EJSON.stringify()`` provides formatting options that reduce the need
+for an additional JSON parser like ``jq``. 
+
+.. code-block:: shell
+
+   # Note: This example is formatted to fit on the page.
+
+   mongosh --quiet \
+           --eval "EJSON.stringify( \
+                db.sales.find( {}, \
+                   { _id: 0, custId: 1, quantity: 1 } ).toArray(), null, 2 \
+             );"
+
+The output is: 
+
+.. code-block:: json
+   :copyable: false 
+
+   [
+     {
+       "custId": 345,
+       "quantity": 4
+     },
+     {
+       "custId": 346,
+       "quantity": 3
+     },
+     {
+       "custId": 486,
+       "quantity": 9
+     }
+   ]
+
+Legacy ``tojsononeline()``
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``mongosh`` formats query output differently than the deprecated
```

Comment:
[suggestion]

Consider removing "The" here.

Reply:
Done